### PR TITLE
feat: adapt "BranchManager" with OpenAPIv2

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2k67b7u Manager 对接：Branch](https://app.clickup.com/t/2k67b7u)

File: graviti/manager/branch.py [hunk 1]
```diff
@@ -61,31 +59,32 @@ class BranchManager:
...
 
     """
 
     def __init__(self, dataset: "Dataset") -> None:
         self._dataset = dataset
-
-    def _generate(
-        self, name: Optional[str] = None, offset: int = 0, limit: int = 128
-    ) -> Generator[TensorbayBranch, None, int]:
+        self._commonly_used_params = {
```
Comment (linjiX) [inline graviti/manager/branch.py:68]: why group the params into a dict?

Comment (graczhual) [reply graviti/manager/branch.py:68]: These parameters are used in almost every function. Putting them into a dict reduces code duplication.

File: graviti/manager/branch.py [hunk 2]
```diff
@@ -8,16 +8,14 @@
 from functools import partial
```
Comment (graczhual) [inline graviti/manager/branch.py:8]: info is not a very good parameter name

File: graviti/manager/dataset.py [hunk 3]
```diff
@@ -21,11 +21,30 @@
...
+        dataset: The name of the dataset, unique for a user.
+
+    """
+
+    def __init__(self, access_key: str, url: str, owner: str, dataset: str) -> None:
+        self.access_key = access_key
+        self.url = url
+        self.owner = owner
+        self.dataset = dataset
+        self._data = {"access_key": access_key, "url": url, "owner": owner, "dataset": dataset}
```
Comment (linjiX) [inline graviti/manager/dataset.py:45]: all the info data are stored twice, very bad

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 4]
```diff
@@ -26,6 +26,36 @@
...
+        return 4
+
+    def __getitem__(self, key: str) -> str:
+        try:
+            return getattr(self, key)  # type: ignore[no-any-return]
+        except AttributeError:
+            raise IndexError(f"'{key}'") from None
+
+    def __iter__(self) -> Iterator[str]:
+        yield from ("access_key", "url", "owner", "dataset")
```
Comment (linjiX) [inline graviti/manager/dataset.py:56]: do we need to set ReprMixin as the base of DatasetAccessInfo. then we can shall this tuple here and `_repr_attrs`.

File: graviti/manager/branch.py [hunk 5]
```diff
@@ -61,31 +59,28 @@ class BranchManager:
...
     Arguments:
-        dataset: :class:`~graviti.dataset.dataset.Dataset` instance.
+        dataset_access_info: :class:`~graviti.manager.dataset.DatasetAccessInfo` instance.
+        commit_id: The commit id.
 
     """
 
-    def __init__(self, dataset: "Dataset") -> None:
-        self._dataset = dataset
+    def __init__(self, dataset_access_info: "DatasetAccessInfo", commit_id: str) -> None:
```
Comment (linjiX) [inline graviti/manager/branch.py:67]: can we just use `access_info` here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 6]
```diff
@@ -26,6 +26,43 @@
...
+    )
+
+    def __init__(self, access_key: str, url: str, owner: str, dataset: str) -> None:
+        self.access_key = access_key
+        self.url = url
+        self.owner = owner
+        self.dataset = dataset
+
+    def __len__(self) -> int:
+        return 4
```
Comment (linjiX) [inline graviti/manager/dataset.py:54]: `len(self._repr_attrs)`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/dataset.py [hunk 7]
```diff
@@ -26,6 +26,43 @@
 LazyLists = NestedDict[str, LazyList[Any]]
 
 
+class DatasetAccessInfo(ReprMixin, Mapping[str, str]):
```
Comment (linjiX) [inline graviti/manager/dataset.py:29]: shall we put `ReprMixin` after `Mapping` since `Mapping` is a real base class and `ReprMixin` is a mixin class

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**